### ISSUE TITLE
Auth0 middleware

### ISSUE THREAD

**Issue by mrkurt** [superfly/cdn#35]

**Comment by nprail** [comment]
This would be the same basic idea as #30, right?

**Comment by mrkurt** [comment]
Yep! I think it makes sense to do some popular ones individually (like Google and GitHub) and then support things like auth0 as well.

**Comment by nprail** [comment]
All of those are OAuth 2.0 so it would be super easy to build a configurable base middleware and then some with presets on top of that base. I might have some time to work on this Saturday if you want?
